Python bindings not working due to recent mutex changes

I suspect that https://github.com/robotology/icub-main/pull/618 caused problem to the Python bindings. You can find in the bottom of this comment the compiler output. It seems that the swig autogenerated code tries to copy a recursive mutex, and the type is not copiable.

If I comment all the skin bindings ([here](https://github.com/robotology/icub-main/blob/7ee911aa6020c66107806b6547a62824b9b6a51f/bindings/icub.i#L43-L51), [here](https://github.com/robotology/icub-main/blob/7ee911aa6020c66107806b6547a62824b9b6a51f/bindings/icub.i#L93-L101) and [here](https://github.com/robotology/icub-main/blob/7ee911aa6020c66107806b6547a62824b9b6a51f/bindings/icub.i#L142-L164)), swig is happy again.

cc @pattacini @Nicogene 

<details>
<summary>Build error</summary>

```
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:17373:3: warning: delete called on 'iCub::iKin::iKinIterateCallback' that is abstract but has non-virtual destructor [-Wdelete-abstract-non-virtual-dtor]
  delete arg1;
  ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:27481:3: warning: delete called on 'iCub::ctrl::ff2LayNN' that is abstract but has non-virtual destructor [-Wdelete-abstract-non-virtual-dtor]
  delete arg1;
  ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:32412:12: error: overload resolution selected deleted operator '='
      arg2 = *temp;
      ~~~~ ^ ~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/mutex:101:22: note: candidate function has been explicitly deleted
    recursive_mutex& operator=(const recursive_mutex&) = delete;
                     ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:32416:35: error: overload resolution selected deleted operator '='
  if (arg1) (arg1)->recursive_mtx = arg2;
            ~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/mutex:101:22: note: candidate function has been explicitly deleted
    recursive_mutex& operator=(const recursive_mutex&) = delete;
                     ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:32438:10: error: overload resolution selected deleted operator '='
  result =  ((arg1)->recursive_mtx);
  ~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/mutex:101:22: note: candidate function has been explicitly deleted
    recursive_mutex& operator=(const recursive_mutex&) = delete;
                     ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:32439:39: error: call to deleted constructor of 'std::recursive_mutex'
  resultobj = SWIG_NewPointerObj((new std::recursive_mutex(static_cast< const std::recursive_mutex& >(result))), SWIGTYPE_p_std__recursive_mutex, SWIG_POINTER_OWN |  0 );
                                      ^                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:1187:89: note: expanded from macro 'SWIG_NewPointerObj'
#define SWIG_NewPointerObj(ptr, type, flags)            SWIG_Python_NewPointerObj(NULL, ptr, type, flags)
                                                                                        ^~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/mutex:100:5: note: 'recursive_mutex' has been explicitly marked deleted here
    recursive_mutex(const recursive_mutex&) = delete;
    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:41883:13: warning: 'Mutex' is deprecated: Use std::mutex instead [-Wdeprecated-declarations]
  yarp::os::Mutex *arg2 = (yarp::os::Mutex *) 0 ;
            ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/Mutex.h:34:7: note: 'Mutex' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API_MSG("Use std::mutex instead") Mutex
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:23:43: note: expanded from macro 'YARP_os_DEPRECATED_API_MSG'
#    define YARP_os_DEPRECATED_API_MSG(X) YARP_DEPRECATED_API_MSG(X)
                                          ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:131:38: note: expanded from macro 'YARP_DEPRECATED_API_MSG'
#  define YARP_DEPRECATED_API_MSG(X) YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(X) YARP_os_API
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:117:69: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC_MSG'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(MSG) __attribute__((__deprecated__(MSG)))
                                                                    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:41883:38: warning: 'Mutex' is deprecated: Use std::mutex instead [-Wdeprecated-declarations]
  yarp::os::Mutex *arg2 = (yarp::os::Mutex *) 0 ;
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/Mutex.h:34:7: note: 'Mutex' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API_MSG("Use std::mutex instead") Mutex
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:23:43: note: expanded from macro 'YARP_os_DEPRECATED_API_MSG'
#    define YARP_os_DEPRECATED_API_MSG(X) YARP_DEPRECATED_API_MSG(X)
                                          ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:131:38: note: expanded from macro 'YARP_DEPRECATED_API_MSG'
#  define YARP_DEPRECATED_API_MSG(X) YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(X) YARP_os_API
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:117:69: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC_MSG'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(MSG) __attribute__((__deprecated__(MSG)))
                                                                    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:41902:38: warning: 'Mutex' is deprecated: Use std::mutex instead [-Wdeprecated-declarations]
  arg2 = reinterpret_cast< yarp::os::Mutex * >(argp2);
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/Mutex.h:34:7: note: 'Mutex' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API_MSG("Use std::mutex instead") Mutex
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:23:43: note: expanded from macro 'YARP_os_DEPRECATED_API_MSG'
#    define YARP_os_DEPRECATED_API_MSG(X) YARP_DEPRECATED_API_MSG(X)
                                          ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:131:38: note: expanded from macro 'YARP_DEPRECATED_API_MSG'
#  define YARP_DEPRECATED_API_MSG(X) YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(X) YARP_os_API
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:117:69: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC_MSG'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(MSG) __attribute__((__deprecated__(MSG)))
                                                                    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:41903:26: warning: 'setCallbackLock' is deprecated: Use setCallbackLock with std::mutex instead [-Wdeprecated-declarations]
  result = (bool)(arg1)->setCallbackLock(arg2);
                         ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/BufferedPort.h:289:5: note: 'setCallbackLock' has been explicitly marked deprecated here
    YARP_DEPRECATED_MSG("Use setCallbackLock with std::mutex instead")
    ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/compiler.h:2608:42: note: expanded from macro 'YARP_DEPRECATED_MSG'
#      define YARP_DEPRECATED_MSG(MSG) [[deprecated(MSG)]]
                                         ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:44487:13: warning: 'Mutex' is deprecated: Use std::mutex instead [-Wdeprecated-declarations]
  yarp::os::Mutex *arg2 = (yarp::os::Mutex *) 0 ;
            ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/Mutex.h:34:7: note: 'Mutex' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API_MSG("Use std::mutex instead") Mutex
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:23:43: note: expanded from macro 'YARP_os_DEPRECATED_API_MSG'
#    define YARP_os_DEPRECATED_API_MSG(X) YARP_DEPRECATED_API_MSG(X)
                                          ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:131:38: note: expanded from macro 'YARP_DEPRECATED_API_MSG'
#  define YARP_DEPRECATED_API_MSG(X) YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(X) YARP_os_API
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:117:69: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC_MSG'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(MSG) __attribute__((__deprecated__(MSG)))
                                                                    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:44487:38: warning: 'Mutex' is deprecated: Use std::mutex instead [-Wdeprecated-declarations]
  yarp::os::Mutex *arg2 = (yarp::os::Mutex *) 0 ;
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/Mutex.h:34:7: note: 'Mutex' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API_MSG("Use std::mutex instead") Mutex
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:23:43: note: expanded from macro 'YARP_os_DEPRECATED_API_MSG'
#    define YARP_os_DEPRECATED_API_MSG(X) YARP_DEPRECATED_API_MSG(X)
                                          ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:131:38: note: expanded from macro 'YARP_DEPRECATED_API_MSG'
#  define YARP_DEPRECATED_API_MSG(X) YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(X) YARP_os_API
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:117:69: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC_MSG'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(MSG) __attribute__((__deprecated__(MSG)))
                                                                    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:44506:38: warning: 'Mutex' is deprecated: Use std::mutex instead [-Wdeprecated-declarations]
  arg2 = reinterpret_cast< yarp::os::Mutex * >(argp2);
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/Mutex.h:34:7: note: 'Mutex' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API_MSG("Use std::mutex instead") Mutex
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:23:43: note: expanded from macro 'YARP_os_DEPRECATED_API_MSG'
#    define YARP_os_DEPRECATED_API_MSG(X) YARP_DEPRECATED_API_MSG(X)
                                          ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:131:38: note: expanded from macro 'YARP_DEPRECATED_API_MSG'
#  define YARP_DEPRECATED_API_MSG(X) YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(X) YARP_os_API
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:117:69: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC_MSG'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(MSG) __attribute__((__deprecated__(MSG)))
                                                                    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:44507:26: warning: 'setCallbackLock' is deprecated: Use setCallbackLock with std::mutex instead [-Wdeprecated-declarations]
  result = (bool)(arg1)->setCallbackLock(arg2);
                         ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/BufferedPort.h:289:5: note: 'setCallbackLock' has been explicitly marked deprecated here
    YARP_DEPRECATED_MSG("Use setCallbackLock with std::mutex instead")
    ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/compiler.h:2608:42: note: expanded from macro 'YARP_DEPRECATED_MSG'
#      define YARP_DEPRECATED_MSG(MSG) [[deprecated(MSG)]]
                                         ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:46088:63: warning: 'RateThread' is deprecated: Use PeriodicThread instead [-Wdeprecated-declarations]
    return (void *)((yarp::os::PeriodicThread *)  ((yarp::os::RateThread *) x));
                                                              ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/RateThread.h:38:7: note: 'RateThread' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API_MSG("Use PeriodicThread instead") RateThread : private PeriodicThread
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:23:43: note: expanded from macro 'YARP_os_DEPRECATED_API_MSG'
#    define YARP_os_DEPRECATED_API_MSG(X) YARP_DEPRECATED_API_MSG(X)
                                          ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:131:38: note: expanded from macro 'YARP_DEPRECATED_API_MSG'
#  define YARP_DEPRECATED_API_MSG(X) YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(X) YARP_os_API
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:117:69: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC_MSG'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(MSG) __attribute__((__deprecated__(MSG)))
                                                                    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:46091:63: warning: 'SystemRateThread' is deprecated: Use PeriodicThread(..., == ShouldUseSystemClock::Yes) instead [-Wdeprecated-declarations]
    return (void *)((yarp::os::PeriodicThread *)  ((yarp::os::SystemRateThread *) x));
                                                              ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/RateThread.h:227:7: note: 'SystemRateThread' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API_MSG("Use PeriodicThread(..., == ShouldUseSystemClock::Yes) instead") SystemRateThread : public PeriodicThread
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:23:43: note: expanded from macro 'YARP_os_DEPRECATED_API_MSG'
#    define YARP_os_DEPRECATED_API_MSG(X) YARP_DEPRECATED_API_MSG(X)
                                          ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:131:38: note: expanded from macro 'YARP_DEPRECATED_API_MSG'
#  define YARP_DEPRECATED_API_MSG(X) YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(X) YARP_os_API
                                     ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:117:69: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC_MSG'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC_MSG(MSG) __attribute__((__deprecated__(MSG)))
                                                                    ^
/iit/sources/robotology-superbuild/build/robotology/ICUB/bindings/icubPYTHON_wrap.cxx:46094:63: warning: 'RateThreadWrapper' is deprecated [-Wdeprecated-declarations]
    return (void *)((yarp::os::PeriodicThread *)  ((yarp::os::RateThreadWrapper *) x));
                                                              ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/RateThread.h:243:7: note: 'RateThreadWrapper' has been explicitly marked deprecated here
class YARP_os_DEPRECATED_API RateThreadWrapper : public PeriodicThread
      ^
/iit/sources/robotology-superbuild/build/install/include/yarp/os/api.h:22:36: note: expanded from macro 'YARP_os_DEPRECATED_API'
#    define YARP_os_DEPRECATED_API YARP_DEPRECATED_API
                                   ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:130:31: note: expanded from macro 'YARP_DEPRECATED_API'
#  define YARP_DEPRECATED_API YARP_DEPRECATED_COMPILER_SPECIFIC YARP_os_API
                              ^
/iit/sources/robotology-superbuild/build/install/include/yarp/conf/api.h:116:60: note: expanded from macro 'YARP_DEPRECATED_COMPILER_SPECIFIC'
#  define YARP_DEPRECATED_COMPILER_SPECIFIC __attribute__((__deprecated__))
                                                           ^
13 warnings and 4 errors generated.
bindings/CMakeFiles/icub.dir/build.make:62: recipe for target 'bindings/CMakeFiles/icub.dir/icubPYTHON_wrap.cxx.o' failed
make[2]: *** [bindings/CMakeFiles/icub.dir/icubPYTHON_wrap.cxx.o] Error 1
CMakeFiles/Makefile2:4401: recipe for target 'bindings/CMakeFiles/icub.dir/all' failed
make[1]: *** [bindings/CMakeFiles/icub.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

</details>

--- a/bindings/icub.i
+++ b/bindings/icub.i
@@ -92,6 +92,7 @@ using namespace yarp::sig;
 // skinDynLib
 %include <iCub/skinDynLib/common.h>
 %include <iCub/skinDynLib/Taxel.h>
+%ignore  iCub::skinDynLib::skinPartBase::recursive_mtx;
 %include <iCub/skinDynLib/skinPart.h>
 %include <iCub/skinDynLib/iCubSkin.h>
 %include <iCub/skinDynLib/dynContact.h>

